test-network-function/networking/suite.go

```diff
@@ -324,3 +348,115 @@ func testNodePort(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func parseVariables(res string, declaredPorts map[key]string) error {
+	var p Port
+	err := json.Unmarshal([]byte(res), &p)
+	if err != nil {
+		return err
+	}
+	for element := range p {
+		var k key
+		k.port = p[element].ContainerPort
+		k.protocol = p[element].Protocol
+		declaredPorts[k] = p[element].Name
+	}
+	return nil
+}
+func declaredPortList(container int, podName, podNamespace string, declaredPorts map[key]string) error {
+	ocCommandToExecute := fmt.Sprintf(commandportdeclared, podName, podNamespace, container)
+	res, _ := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(false))
+	err := parseVariables(res, declaredPorts)
+	if err != nil {
+		log.Error(err)
+	}
+	return err
+}
+
+func listeningPortList(commandlisten []string, nodeOc *interactive.Context, listeningPorts map[key]string) error {
+	var k key
+	listeningPortCommand := strings.Join(commandlisten, " ")
+	res, err := utils.ExecuteCommand(listeningPortCommand, ocCommandTimeOut, nodeOc)
+	if err != nil {
+		return err
+	}
+	lines := strings.Split(res, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if indexprotocolname > len(fields) || indexport > len(fields) {
+			return err
+		}
+		s := strings.Split(fields[indexport], ":")
+		p, _ := strconv.Atoi(s[1])
+		k.port = p
+		k.protocol = strings.ToUpper(fields[indexprotocolname])
+		listeningPorts[k] = ""
+	}
+	return nil
+}
+
+func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]string) map[key]string {
+	res := make(map[key]string)
+	if len(listeningPorts) == 0 || len(declaredPorts) == 0 {
+		return res
+	}
+	for k := range listeningPorts {
+		_, ok := declaredPorts[k]
+		if !ok {
+			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
+			res[k] = listeningPorts[k]
+		}
+	}
+	return res
+}
+
+func testListenAndDeclared(env *config.TestEnvironment) {
+	declaredPorts := make(map[key]string)
+	listeningPorts := make(map[key]string)
+	undeclaredPorts := make(map[key]string)
+	var skippedPods []string
```

Comment:
we don't use the content of these lists now, just the length. so it's kind of ok. but it really should just be []Pod instead of []string.

Reply:
you're right! I fixed.